src/prefect/transactions.py

```diff
@@ -81,11 +81,22 @@ def set(self, name: str, value: Any) -> None:
         self._stored_values[name] = value
 
     def get(self, name: str, default: Any = NotSet) -> Any:
-        if name not in self._stored_values:
+        value = self._stored_values.get(name, NotSet)
+        if value is NotSet:
+            parent = self.get_parent()
+            if parent is not None:
+                try:
+                    value = copy.deepcopy(parent.get(name))
+                    self._stored_values[name] = value
```

Comment:
Yeah, you can see the use case I was thinking of in the test. If I get and mutate a value (e.g. append to a list) I'd expect to see that update the next time I get the value. That makes storing the value on the child transaction necessary.

Reply:
ok interesting, but there's an analogous situation that could be equally argued for in which the parent mutates the value after the child calls `get`; only sharing that because the lack of obvious priority there suggests to me that there's a better interface for getting data into transaction hooks.  Should we consider allowing for the same signature as state change hooks?

Reply:
in the meantime, i'm OK with the version you have here

Reply:
That's a good point about the order still mattering. What if said that you can only update a stored value in a transaction by calling `.set()`? Mutations to values returned by `.get` wouldn't propagate the stored values. That seems like it'd be more predictable.

Reply:
Yea I really like that; in that case we'd always return a `deepcopy` from `get`, right?

Reply:
Yeah that'd be necessary to make the stored values "immutable"

Reply:
@cicdw I've made the update to return copies of stored values and it feels pretty good. Let me know what you think!